core/trino-spi/pom.xml

```diff
@@ -129,60 +129,68 @@
-                                if (!match.matches()) {
-                                    throw new RuntimeException("Invalid version: " + project.version)
-                                }
-                                int version = (match.group(1) as int) - 1
-                                project.properties.setProperty('releasedVersion', version as String)
-                            </source>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-dependency-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>copy</id>
-                        <phase>generate-test-resources</phase>
-                        <goals>
-                            <goal>copy</goal>
-                        </goals>
-                        <configuration>
-                            <skip>false</skip>
-                            <artifactItems>
-                                <artifactItem>
-                                    <groupId>io.trino</groupId>
-                                    <artifactId>trino-spi</artifactId>
-                                    <version>${releasedVersion}</version>
-                                    <type>jar</type>
-                                    <outputDirectory>${released-artifacts.dir}</outputDirectory>
-                                </artifactItem>
-                                <artifactItem>
-                                    <!-- slice is needed when loading trino-spi classes for backward compatibility tests -->
-                                    <groupId>io.airlift</groupId>
-                                    <artifactId>slice</artifactId>
-                                    <type>jar</type>
-                                    <outputDirectory>${released-artifacts.dir}</outputDirectory>
-                                </artifactItem>
-                            </artifactItems>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <ignoreSuggestionsFormat>xml</ignoreSuggestionsFormat>
+                    <analysisConfiguration>
+                        <revapi.filter>
+                            <elements>
+                                <exclude>
+                                    <item>
+                                        <matcher>java</matcher>
+                                        <match>@io.trino.spi.Experimental(*) ^*;</match>
+                                    </item>
+                                </exclude>
+                            </elements>
+                        </revapi.filter>
+                        <revapi.differences>
```

Comment:
Can this go into separate file?
The separate file is easier to read and makes pom easier to read (the list will grow)

In Iceberg's revapi usage there is also a command to update the allowed breakages list, which i assume is easier with a separate file.

Reply:
The list will only contain breakages for the previous release, since we only compare against that one. After each release, we could delete any exclusions. I can update the comment to make that more clear. The items above the comment are permanent configuration.

When the checker fails, it outputs the XML to copy/paste into the POM. Breakages should be rare with the new `@Experimental` annotation. It's certainly easier than today, which requires editing strings in Java source code.

While I'm not completely against using a separate file, I'm also not convinced it's worthwhile to investigate now. This seems like a big improvement over what we have today, and we can see how it plays out over a few releases and iterate if needed.

Reply:
I updated the comment and also added explanations for the permanent exclusions.
```xml
<!-- Backwards incompatible changes since the previous release (393) -->
<!-- Any exclusions below can be deleted after each release -->